crates/biome_js_analyze/src/lint/style/use_naming_convention.rs

```diff
@@ -278,10 +301,170 @@ declare_rule! {
+    ///
+    /// If a declaration is not selected or if a capture is forwarded while there is no more custom conventions,
+    /// Then the declaration is verified against the default convention.
+    /// If a forwarded capture is a part of the original name, then underscore and dollar signs are not trimmed.
+    ///
+    /// In the following example:
+    ///
+    /// = We require `static readonly` class members to be in ["CONSTANT_CASE"].
+    /// - We require `private` class members to start with an underscore `_` and to be in [`camelCase`].
+    /// - We require global constants to be in ["CONSTANT_CASE"] and
+    ///   we allow these constants to be enclosed by double underscores or to be named `_SPECIAL_`.
+    /// - We require interfaces to start with `I`, except for interfaces ending with `Error`,
+    ///   and to be in [`PascalCase`].
+    /// - All other names follow the default conventions
+    ///
+    ///```jsonc
+    /// {
+    ///     "//": "...",
+    ///     "options": {
+    ///         "custon": [
+    ///             {
+    ///                 "selector": {
+    ///                     "kind": "classMember",
+    ///                     "modifiers": ["static", "readonly"]
+    ///                 },
+    ///                 "formats": ["CONSTANT_CASE"]
+    ///             },
+    ///             {
+    ///                 "selector": {
+    ///                     "kind": "classMember",
+    ///                     "modifiers": ["private"]
+    ///                 },
+    ///                 "match": "_(.+)",
+    ///                 "formats": ["camelCase"]
+    ///             }, {
+    ///                 "selector": {
+    ///                     "kind": "const",
+    ///                     "scope": "global"
+    ///                 },
+    ///                 "match": "__(.+)__|_SPECIAL_|(.+)",
+    ///                 "formats": ["CONSTANT_CASE"]
+    ///             }, {
+    ///                 "selector": {
+    ///                     "kind": "interface"
+    ///                 },
+    ///                 "match": "I(.*)|(.*)Error",
+    ///                 "formats": ["PascalCase"]
+    ///             }
+    ///             // default conventions
+    ///         ]
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ### Regular expression syntax
+    ///
+    /// The `match` option takes a regular expression that supports the following syntaxes:
+    ///
+    /// - Greedy quantifiers `*`, `?`, `+`, `{n}`, `{n,m}`, `{n,}`, `{m}`
+    /// - Non-greedy quantifiers `*?`, `??`, `+?`, `{n}?`, `{n,m}?`, `{n,}?`, `{m}?`
+    /// - Any character matcher `.`
+    /// - Character classes `[a-z]`, `[xyz]`, `[^a-z]`
+    /// - Alternations `|`
+    /// - Capturing groups `()`
+    /// - Non-capturing groups `(?:)`
+    /// - A limited set of escaped characters including all special characters
+    ///   and regular string escape characters `\f`, `\n`, `\r`, `\t`, `\v`
```

Comment:
I think we should move this part inside our website, maybe in the analyzer or another section of the website, mostly because this explanation feels reductive. I think we should be more thorough and provide examples for each bullet point where we explain the patterns

Reply:
You are right. For now, only this rule uses regexes, however this could change in the future. I will remove this and open a PR on the website repository.

Reply:
You are right. For now, only this rule uses regexes, however this could change in the future.

I will open a PR on the website repository.
I am unsure where this section should be.
Thus, I will keep this overview for now and will remove it in another PR with a proper link to the future regex section on the website.